segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/NoOpStorage.java

```diff
@@ -0,0 +1,265 @@
+import lombok.extern.slf4j.Slf4j;
+
+import static io.pravega.shared.NameUtils.INTERNAL_SCOPE_NAME;
+
+/**
+ * Storage adapter for testing scenario without interference from tier 2.
+ *
+ * All the writes to tier 2 are no-oped unless its for metadata (system segments).
+ * Reads are not supported.
+ * Can also add latency to write operation to make the no-oped write as close to real as possible.
+ */
+@Slf4j
+class NoOpStorage implements SyncStorage {
+
+    //Hidden in memory storage to verify the correctness of no-operation.
+    private final SyncStorage noopInMemoryStorage;
+
+    private final SyncStorage baseStorage;
+    private final int writeNoOpLatencyMill;
+
+    /**
+     * Creates a new instance of the NoOpStorage class.
+     *
+     * @param config   The configuration to use.
+     */
+    NoOpStorage(StorageExtraConfig config, SyncStorage baseStorage) {
+        Preconditions.checkNotNull(config, "config");
+        this.writeNoOpLatencyMill = config.getStorageWriteNoOpLatencyMill();
+        this.baseStorage = Preconditions.checkNotNull(baseStorage, "baseStorage");
+
+        noopInMemoryStorage = new InMemoryStorage();
+    }
+
+    @Override
+    public void close() {
+        baseStorage.close();
+        if (noopInMemoryStorage != null) {
+            noopInMemoryStorage.close();
+        }
+    }
+
+    @Override
+    public void initialize(long epoch) {
+        baseStorage.initialize(epoch);
+        if (noopInMemoryStorage != null) {
+            noopInMemoryStorage.initialize(1);
+        }
+    }
+
+    @Override
+    public SegmentProperties getStreamSegmentInfo(String streamSegmentName) throws StreamSegmentException {
+        if (isSystemSegment(streamSegmentName)) {
+            return baseStorage.getStreamSegmentInfo(streamSegmentName);
+        } else {
+            if (noopInMemoryStorage != null) {
+                return noopInMemoryStorage.getStreamSegmentInfo(streamSegmentName);
+            } else {
+                throw new UnsupportedOperationException("getStreamSegmentInfo() for non-system segment is not supported in NO-OP mode.");
+            }
+        }
+    }
+
+    @Override
+    public boolean exists(String streamSegmentName) {
+        if (isSystemSegment(streamSegmentName)) {
+            return baseStorage.exists(streamSegmentName);
+        } else {
+            if (noopInMemoryStorage != null) {
+                return noopInMemoryStorage.exists(streamSegmentName);
+            } else {
+                throw new UnsupportedOperationException("exists() for non-system segment is not supported in NO-OP mode.");
+            }
+        }
+    }
+
+    @Override
+    public int read(SegmentHandle handle, long offset, byte[] buffer, int bufferOffset, int length) throws StreamSegmentException {
+        if (isSystemSegment(handle.getSegmentName())) {
+            return baseStorage.read(handle, offset, buffer, bufferOffset, length);
+        } else {
+            if (noopInMemoryStorage != null) {
```

Comment:
This if statement is in every single method in this class. Perhaps you should just make two classes that have the same interface.

Reply:
fixed